File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F19.P2.swift
// BEGIN PROBE F19.P2
import Synchronization

let counter = Mutex<Int64>(0)
await withTaskGroup { group in
    for _ in 0..<2 {
        group.addTask {
            for _ in 0..<1000 { counter.withLock { $0 += 1 } }
        }
    }
}
let total = counter.withLock { $0 }
// END PROBE F19.P2

print(total)

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F19.P2.go
package main

import "fmt"

// BEGIN PROBE F19.P2
import (
	"sync"
	"sync/atomic"
)

func raceFree() int64 {
	var counter int64
	var wg sync.WaitGroup
	for range 2 {
		wg.Go(func() {
			for range 1000 {
				atomic.AddInt64(&counter, 1)
			}
		})
	}
	wg.Wait()
	return counter
	// END PROBE F19.P2
}

func main() {
	fmt.Println(raceFree())
}
